cgi/nutrients.pl

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2019 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use Modern::Perl '2012';
+use utf8;
+
+use CGI::Carp qw(fatalsToBrowser);
+use CGI qw/:cgi :form escapeHTML/;
+
+use ProductOpener::Display qw/:all/;
+use ProductOpener::Food qw/:all/;
+
+use CGI qw/:cgi :form escapeHTML/;
+use JSON::PP;
+
+ProductOpener::Display::init();
+
+# Recursively remove parent association to avoid redundant JSON data.
+sub _remove_parent {
+	my $current_ref = shift;
+
+	if (defined $current_ref->{nutrients}) {
+		foreach my $nutrient (@{$current_ref->{nutrients}}) {
+			_remove_parent($nutrient);
+		}
+	}
+
+	delete $current_ref->{parent};
+}
+
+my @table = ();
+my $previous_ref;
+my $previous_prefix_length = 0;
+foreach (@{$nutriments_tables{$nutriment_table}}) {
```

Comment:
I don't see $nutriment_table defined?

Reply:
That's initialized by `ProductOpener::Display::init`
https://github.com/openfoodfacts/openfoodfacts-server/blob/c2d74a1f9a9b4d0fe853d0c8f0396bf0eaf07e1f/lib/ProductOpener/Display.pm#L342-L346